+adi/+FMComms5/Tx.m

```diff
@@ -0,0 +1,356 @@
+                end
+            end
+            %% Set scales
+            s = size(obj.DDSScalesChipB);
+            indx = 0;
+            for channel=1:s(2)
+                for toneIdx = 1:2
+                    id = sprintf('altvoltage%d',indx);
+                    indx = indx + 1;
+                    chanPtr = getChan(obj,obj.iioDevChipB,id,true);
+                    iio_channel_attr_write_double(obj,chanPtr,'scale',obj.DDSScalesChipB(toneIdx,channel));
+                end
+            end
+            %% Set phases
+            s = size(obj.DDSPhasesChipB);
+            indx = 0;
+            for channel=1:s(2)
+                for toneIdx = 1:2
+                    id = sprintf('altvoltage%d',indx);
+                    indx = indx + 1;
+                    chanPtr = getChan(obj,obj.iioDevChipB,id,true);
+                    iio_channel_attr_write_double(obj,chanPtr,'phase',obj.DDSPhasesChipB(toneIdx,channel));
+                end
+            end
+        end
+        
+        function setupInit(obj)
+            obj.iioDevChipB = getDev(obj, obj.devNameChipB);
+            
+            % Write all attributes to device once connected through set
+            % methods
+            setupLibad9361(obj);
+            obj.iioDevPHYChipB = calllib('libiio', 'iio_context_find_device',obj.iioCtx,'ad9361-phy-B');
+            
+            % Do writes directly to hardware without using set methods.
+            % This is required sine Simulink support doesn't support
+            % modification to nontunable variables at SetupImpl
+            id = 'altvoltage1';
+            obj.setAttributeLongLong(id,'frequency',obj.CenterFrequency,true,8);
+            obj.setAttributeLongLong(id,'frequency',obj.CenterFrequencyChipB,true,8,obj.iioDevPHYChipB);
+            if  ~obj.EnableCustomFilter
+                if libisloaded('libad9361')
+                    calllib('libad9361','ad9361_set_bb_rate',obj.iioDevPHY,int32(obj.SamplingRate));
+                    calllib('libad9361','ad9361_set_bb_rate',obj.iioDevPHYChipB,int32(obj.SamplingRate));
+                else
+                    obj.setAttributeLongLong('voltage0','sampling_frequency',obj.SamplingRate,true,4);
+                    obj.setAttributeLongLong('voltage0','rf_bandwidth',obj.RFBandwidth,strcmp(obj.Type,'Tx'));
+                    obj.setAttributeLongLong('voltage0','sampling_frequency',obj.SamplingRate,true,4,obj.iioDevPHYChipB);
+                    obj.setAttributeLongLong('voltage0','rf_bandwidth',obj.RFBandwidthChipB,strcmp(obj.Type,'Tx'),[],obj.iioDevPHYChipB);
+                end
+            else
+                writeFilterFile(obj);
+                writeFilterFileFMComms5ChipB(obj);
+            end            
+            
+            obj.setAttributeLongLong('voltage0','hardwaregain',obj.AttenuationChannel0,true);
+            if (obj.channelCount>2)
```

Comment:
Same as RX, this will lead to gains not set cases like [3,4]==EnabledChannels

Reply:
Addressed by the commit.